rest/specs/src/main/filtered-resources/openapi/openapi.yaml

```diff
@@ -54,26 +117,133 @@ paths:
           $ref: '#/components/responses/NotImplemented'
         default:
           $ref: '#/components/responses/DefaultError'
-
+  /transform/{TransformId}:
+    parameters:
+      - $ref: '#/components/parameters/TransformId'
+    get:
+      summary: Poll transformation request status
+      description: >
+        After posting a transformation request, the client can use the returned URI to poll for the transform's completion at this endpoint.
+      operationId: poll
+      tags:
+        - transform
+      responses:
+        '200':
+          description: The transform is still in progress.
+          headers:
+            Content-Version:
+              $ref: '#/components/headers/ContentVersion'
+        '201':
```

Comment:
Several resources say that a `303 See Other` should be returned here, redirecting to a different endpoint that will actually return the metadata URI's.
https://farazdagi.com/2014/rest-and-long-running-jobs/
http://restalk-patterns.org/long-running-operation-polling.html
I felt that that was overkill since we are not permanently storing this metadata, so I went with the simpler approach laid out here
https://stackoverflow.com/a/9527394

Reply:
If the transform service can understand the request and decide if the job is in-process, failed, or succeeded, then return a 200. Give the status of the transform job, like "in-progress", "failed", or "complete" in the message body or in a response header (e.g. `X-Job-Status`) .

Reply:
Well, as it is now, the idea is that if it's "in progress" we return a 200, if it's "failed" we return a 500 or something similar, if it's "complete" we return a 201. I'm not sure that we want to get into providing specific status's without using different status codes for each of them, do we?

Reply:
I am personally not a fan of returning 200 if it is in progress. I much prefer having a different return code that is not a 2xx so we can clearly differentiate the final and successful response (i.e. 201) from a failure (i.e. 500) from something indicating it is still in progress (like 100)

Reply:
@brjeter. Those status codes refer to the request for for status and not the status of the transform job. Here is how I imagine it:

Response Code | Meaning | Response Body
--------------------- | ------------ | ----------------------
404 | Transform ID not found | None
500 | Unable to determine transform status | None
200 | Transform status available | {status: INPROGRESS}
200 | Transform status available | {status: COMPLETE}
200 | Transform status available | {status: FAILED}

Reply:
@ahoffer I think we understand what you're proposing. We're just not sure that it's a good idea to stray away from relying on the HTTP status codes as much as possible, per REST standards. There's a lot of implications to using these body-embedded statuses instead of HTTP codes, but I think the biggest tangible one is that the client has to implement more custom logic to handle these body-embedded statuses rather than the utilities that HTTP libraries already have around handling the common HTTP statuses.

Reply:
https://softwareengineering.stackexchange.com/questions/316208/http-status-code-for-still-processing Describes exactly our case and they have a good suggestion to use http status codes. As we are planning, they suggest on the initial request to return a 202 and return the status URL where to go to next in the Location header. on that URL, they suggest to return 404 or 409 until such time as the request is completed. I am not a fan of the 404 since we will use that later to represent the request no longer exist and we don't want to mix the 2. Not a fan of the 409 either since it typically requires the client to re-issue the request which is not our case. None of the available codes are perfect and I think 102 (https://httpstatuses.com/102) is the closet one to what we need. The fact that it is not a 2xx is good because the request for which we are querying ain't successful yet. It ain't  a 5xx because it hasn't failed yet and it ain't a 4xx which typically means a client error.

Reply:
@ahoffer, again I still think that solution is slightly more complicated, not only in the implementation but also in the API. What are the advantages of your solution that could make that tradeoff worth it?

Reply:
On top of everything, using standard HTTP codes allows for using 3rd party intermediate like proxies, load balancers, caching ...

Reply:
@paouelle 102 does look like a good fit for what the API wants to communicate.

@brjeter Advantage of KV pairs in a JSON body are:
1. Words that are clearer than status codes. 
2. Status codes are extensible.  There is already talk of adding sub-codes, which would have to go in a custom header or body.

@paouelle 
> using standard HTTP codes allows for using 3rd party intermediate like proxies, load balancers, caching
There are times when that is true, but I don't see that here. On top of everything, that fact that there is so much discussion on the PR asking "what is the right code" means it unlikely that a 3rd party will understand what the codes are intended to mean. If it is not clear to us what status codes to use, it will not be clear to a 3rd party trying to divine our intentions. At least, no without carefully reading the documentation we provide.